src/mmw/js/src/draw/views.js

```diff
@@ -201,6 +188,61 @@ var ResetDrawView = Marionette.ItemView.extend({
     }
 });
 
+function getShapeAndAnalyze(e, model, ofg, grid, tableId) {
+    console.log('clicked');
+    // The shapeId might not be available at the time of the click
+    // because the UTF Grid layer might not be loaded yet, so
+    // we poll for it.
+    var pollInterval = 200,
+        maxPolls = 5,
+        pollCount = 0,
+        deferred = $.Deferred(),
+        shapeId = e.data ? e.data.id : null;
+
+        if (shapeId) {
+            _getShapeAndAnalyze();
+        } else {
+            pollForShapeId();
+        }
+
+    function _getShapeAndAnalyze() {
```

Comment:
I don't think the underscore prefix is necessary for this internal function, or, I would keep it consistent with the other internal function.